File: TheSharks.UI/src/types/application/activity/Activity.ts
import { Participant } from "./Participant";

export interface Activity {
    departure?: Date;
    briefingTime?: Date;
    tide?: string;
    atWater?: Date;
    startTime?: Date;
    endTime?: Date;
    id: string;
    activityType: string;
    responsibleId: string;
    responsibleFirstName: string;
    responsibleLastName: string;
    title: string;
    name: string;
    date: Date;
    location: string;
    locationLink: string;
    info: string;
    memberInfo: string;
    necessarySubscription: boolean;
    enrollments: Participant[]
}

File: TheSharks.Domain/Entities/Activity.cs
namespace TheSharks.Domain.Entities;

public abstract class Activity
{
    public Guid Id { get; set; }
    public string ActivityType { get; set; }
    public string Name { get; set; }
    public DateTimeOffset Date { get; set; }
    public string Location { get; set; }
    public string? LocationLink { get; set; }
    public string? Info { get; set; }
    public string? MemberInfo { get; set; }
    public bool NecessarySubscription { get; set; } = false;
    public Member Responsible { get; set; }
    public ICollection<Enrollment> Enrollments { get; set; }
}